Changes around cutomer db upgrade to make it fully automated using rhev

sghai opened the pull request:
Changes to support auto creation of instances on rhevm for customer db upgrade.

sghai commented on scripts/satellite6_upgrader.sh:
```diff
@@ -34,15 +34,22 @@ if [ "${DISTRIBUTION}" = 'DOWNSTREAM' ]; then
+    INSTANCE_NAME="${CUSTOMERDB_NAME}_customerdb_instance"_
+    # Create a RHEV instance of RHEL6/7 with given template, datacenter, quota and cluster
+    if [ "${OS}" = 'rhel7' ]; then
+        SAT_INSTANCE_FQDN=$(fab -u root create_rhevm_instance:"${INSTANCE_NAME}",'custdb-rhel7-base','SAT-QE','SAT-QE','SAT-QE')
```
for now we have defined the image name, datacenter, quota and cluster directly here. We can move to `jenkins-config` later, if needed.

kbidarkar replied:
I think, it is ok , if it is here too, jenkins-config is mostly a "all internal stuff" kind of thing.

sghai commented on scripts/satellite6_upgrader.sh:
```diff
@@ -34,15 +34,22 @@ if [ "${DISTRIBUTION}" = 'DOWNSTREAM' ]; then
+        SAT_INSTANCE_FQDN=$(fab -u root create_rhevm_instance:"${INSTANCE_NAME}", 'custdb-rhel6-base','SAT-QE','SAT-QE','SAT-QE')
+    fi
+    # Clone the 'satellite-clone' project that includes the ansible playbook to install sat server along with customer DB.
+    git clone -b satellite-clone-1.0 https://github.com/RedHatSatellite/satellite-clone.git
```
we don't want to use `master` branch directly due to massive changes. So we used the stable `1.0` branch.

sghai commented:
this PR is in reference w/ `jenkins-configs/merge_requests/41` and `SatelliteQE/automation-tools/pull/504`

kbidarkar commented on scripts/satellite6_upgrader.sh:
```diff
@@ -34,15 +34,22 @@ if [ "${DISTRIBUTION}" = 'DOWNSTREAM' ]; then
-    git clone https://github.com/RedHatSatellite/satellite-clone.git
+if [ "${CUSTOMERDB}" = 'true' ]; then
     source config/preupgrade_entities.conf
+    INSTANCE_NAME="${CUSTOMERDB_NAME}_customerdb_instance"_
```
Looks like a typo, "_" after the quotes.

jyejare replied:
Remove '_' underscore at the end of line.

jyejare commented on scripts/satellite6_upgrader.sh:
```diff
@@ -60,13 +67,17 @@ if [ ! -z "${CUSTOMER_NAME}" ]; then
-    # Customer DB URL
-    DB_URL="http://"${cust_db_server}"/pub/customer-databases/"${CUSTOMER_NAME}""
+    # Prepare Customer DB URL
+    if [ "${CUSTOMERDB_NAME}" = 'Lidl' ]; then
```
Instead of many if-else loops, keep dictionary with mappings where key is a customer name and value is DB url.

kbidarkar replied:
@jyejare, this is a shell script, dictionary with mappings ??

sghai replied:
@jyejare : yeah.. its shell :)

jyejare replied:
I thought there should be similar in Shell. Its ok we can figure out that later.

jyejare commented on jobs/satellite6-upgrader.yaml:
```diff
@@ -62,16 +62,22 @@
                 <p>The number of RHEL6 and RHEL7 clients to Generate.</p>
                 <p>The number of clients will be devided in equal amount for RHEL6 and RHEL7. e.g if 10 is given then 5 each.</p>
+        - bool:
+            name: CUSTOMERDB
```
remove this extra parameter and add 'none' to the below 'CUSTOMERDB_NAME' parameter. 
Then set condition as :
'if CUSTOMERDB_NAME is not none'
wherever required.

kbidarkar replied:
@jyejare, if you try to do that it would be a string "None" and not None.

jyejare replied:
Yeah I am thinking it as string only. Or may be we can have something like '--select--' or just 'select'.
Options would be something like:
1. None/Select
2. Lidl
3. ExpressScripts

and,

if [  'CUSTOMERDB_NAME' != 'None' ]; then
      execute everything here for that perticular customer
fi

jyejare commented on jobs/satellite6-upgrader.yaml:
```diff
@@ -62,16 +62,22 @@
+        - choice:
+            name: CUSTOMERDB_NAME
+            choices:
+                - 'Lidl'
```
Add 1st(default) choice as none

kbidarkar replied:
@jyejare, This would not work, same as suggested above, adding a 'None' here would be considered a string "None" and not None.

jyejare replied:
Commented above.

jyejare commented on scripts/satellite6_upgrader.sh:
```diff
@@ -34,15 +34,22 @@ if [ "${DISTRIBUTION}" = 'DOWNSTREAM' ]; then
-if [ ! -z "${CUSTOMER_NAME}" ]; then
-    # Clone the 'satellite-clone' project that includes the ansible playbook to install sat server along with customer DB.
-    git clone https://github.com/RedHatSatellite/satellite-clone.git
+if [ "${CUSTOMERDB}" = 'true' ]; then
```
Change this to :
if [  "${CUSTOMERDB_NAME}" != 'None' ]; then as per my above comments.

jyejare commented on scripts/satellite6_upgrader.sh:
```diff
@@ -34,15 +34,22 @@ if [ "${DISTRIBUTION}" = 'DOWNSTREAM' ]; then
     source config/preupgrade_entities.conf
+    INSTANCE_NAME="${CUSTOMERDB_NAME}_customerdb_instance"_
+    # Create a RHEV instance of RHEL6/7 with given template, datacenter, quota and cluster
+    if [ "${OS}" = 'rhel7' ]; then
```
Remove if..else block here instead one single line is enough as below :
SAT_INSTANCE_FQDN=$(fab -u root create_rhevm_instance:"${INSTANCE_NAME}",'custdb-"${OS}"-base','SAT-QE','SAT-QE','SAT-QE')

So OS will be always either rhel7 and rhel6. And you can directly insert $OS value.

jyejare commented on scripts/satellite6_upgrader.sh:
```diff
@@ -34,15 +34,22 @@ if [ "${DISTRIBUTION}" = 'DOWNSTREAM' ]; then
+    # Create a RHEV instance of RHEL6/7 with given template, datacenter, quota and cluster
+    if [ "${OS}" = 'rhel7' ]; then
+        SAT_INSTANCE_FQDN=$(fab -u root create_rhevm_instance:"${INSTANCE_NAME}",'custdb-rhel7-base','SAT-QE','SAT-QE','SAT-QE')
+    elif [ "${OS}" = 'rhel6' ]; then
```
Remove this else as per above comment.

kbidarkar commented:
ACK

sghai commented:
@jyejare : could you please take another look ? I addressed comment w/ `${OS}` while creating the instance, not all.  But still would like your thoughts if you'r okay w/ the current changes ?

jyejare commented:
@sghai @kbidarkar , My intention is to reduce as the parameters on the job so that user wont confuse or less efforts required to trigger the job.

sghai commented:
@jyejare: I made the change but I still not sure if adding "None" option is a good idea.. but if it won't confuse user then I'm fine w/ it.

jyejare commented:
@sghai , If you dont like 'None' then just change it to 'Select' which will tell user to select something. :)
Anyhow this not a blocker comment.

I am ACKing.
